File: examples and misc/Python/inheritance.py
# Base class
class Animal:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def speak(self):
        print(f"{self.name} makes a sound.")

# Derived class
class Dog(Animal):
    def speak(self):
        print(f"{self.name} barks.")

# Create an instance of Dog
my_dog = Dog("Buddy", 3)
my_dog.speak()

File: examples and misc/Cpp/inheritance.cpp
#include <iostream>
using namespace std;

// Base class
class Animal {
public:
    string name;
    int age;

    Animal(string n, int a) : name(n), age(a) {}

    void speak() {
        cout << name << " makes a sound." << endl;
    }
};

// Derived class
class Dog : public Animal {
public:
    Dog(string n, int a) : Animal(n, a) {}

    void speak() {
        cout << name << " barks." << endl;
    }
};

int main() {
    Dog myDog("Buddy", 3);
    myDog.speak();
    return 0;
}
